src/modules/csm/checkpoint.py

```diff
@@ -229,18 +234,47 @@ def _prepare_committees(self, epoch: EpochNumber) -> Committees:
 
 def process_attestations(attestations: Iterable[BlockAttestation], committees: Committees) -> None:
     for attestation in attestations:
-        committee_id = (attestation.data.slot, attestation.data.index)
-        committee = committees.get(committee_id, [])
-        att_bits = _to_bits(attestation.aggregation_bits)
-        for index_in_committee, validator_duty in enumerate(committee):
-            validator_duty.included = validator_duty.included or _is_attested(att_bits, index_in_committee)
+        committee_offset = 0
+        for committee_idx in get_committee_indices(attestation):
+            committee = committees.get((attestation.data.slot, committee_idx), [])
+            att_bits = hex_bitlist_to_list(attestation.aggregation_bits)[committee_offset:][: len(committee)]
+            for index_in_committee in get_set_indices(att_bits):
+                committee[index_in_committee].included = True
+            committee_offset += len(committee)
+
+
+def get_committee_indices(attestation: BlockAttestation) -> list[CommitteeIndex]:
+    if not is_electra_attestation(attestation):
+        return [attestation.data.index]
+    return [str(i) for i in get_set_indices(hex_bitvector_to_list(attestation.committee_bits))]
+
+
+def is_electra_attestation(attestation: BlockAttestation) -> TypeGuard[BlockAttestationElectra]:
+    has_committee_bits = getattr(attestation, "committee_bits") is not None
+    has_zero_index = attestation.data.index == "0"
```

Comment:
Does it work the same for v1 and v2 endpoints? Or `attestation.data.index` is removed for v2
Just to be safe in case of a version change in our client

Reply:
When I tested I had the same results, and we use v1 only for now.  `attestation.data.index` is not removed and kept equal to "0" as of spec.